components/filter2/dropdown_group/view.coffee

```diff
@@ -35,7 +35,7 @@ module.exports = class DropdownGroupView extends Backbone.View
 
   toggleForSale: ->
     if @params.get('for_sale') is 'true'
-      @params.set 'for_sale', 'false'
+      @params.set 'for_sale', null
```

Comment:
Lmk if you want to explore this though. There might have been something I missed.